pkg/kube/manifests/secret/secret.go

```diff
@@ -0,0 +1,98 @@
+// Package secret provides a secret creator and applier
+package secret
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// Secret contains the state for creating a secret
+type Secret struct {
+	Name      string
+	Namespace string
+	Manifest  *v1.Secret
+	Ctx       context.Context
+}
+
+// New returns an initialised secret creator
+func New(name, namespace string, manifest *v1.Secret) *Secret {
+	return &Secret{
+		Name:      name,
+		Namespace: namespace,
+		Manifest:  manifest,
+		Ctx:       context.Background(),
+	}
+}
+
+// DeleteSecret deletes the secret
+func (s *Secret) DeleteSecret(_ kubernetes.Interface, config *rest.Config) (interface{}, error) {
+	client := kubernetes.NewForConfigOrDie(config)
```

Comment:
"Or die". I'm assuming this means a panic. As far as I can tell, we've been handling errors by returning them all the way back up the stack everywhere else. Why the change of behaviour?

Reply:
Meltdown? I have no idea, we shouldn't be doing this, changing!